plugins/aws/fix_plugin_aws/resource/base.py

```diff
@@ -220,20 +220,26 @@ def collect_resources(cls: Type[AwsResource], builder: GraphBuilder) -> None:
                 raise
 
     @classmethod
-    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> List[AwsResource]:
```

Comment:
I was on the fence to propose the following:
```python
    def collect(cls: Type[AwsResourceType], json: List[Json], builder: GraphBuilder) -> List[AwsResourceType]:
   buckets = AwsS3Bucket.collect()
```
buckets would have type AwsS3Bucket automatically.
But it would also mean, that each collect method will only collect instances of the related class.
But there are collect methods that also collect other resources.
See IAM user.

As a consequence: when instances of other kinds are collected, they should be returned.